Automattic/jetpack
# Explicit single execution of do_full_sync from cron

In Jetpack 8.3 the Immediate Full Sync process was rolled out to users. With this update it modified the return type of do_full_sync to always be WP_Error. This unexpected behavior caused cron and cli to only do a single iteration vs sending data for their max execution times. 

This update ensures that only a single iteration of the do_full_sync method is allowed when cron is processing Full Sync actions. 

Note that I was working off of https://github.com/Automattic/jetpack/pull/16046 and this can be rebased to master, however the additional tests provide context for states.

See: p1HpG7-9pe-p2

#### Changes proposed in this Pull Request:
* Explicitly restrict cron to only sending a single Full Sync request to WP.com for processing.

#### Is this a new feature or does it add/remove features to an existing part of Jetpack?
* See p1HpG7-9pe-p2 for more details.

#### Does this pull request change what data or activity we track or use?
No change

#### Testing instructions:

Duplicate issue in 8.6.0
* Apply https://github.com/Automattic/jetpack/pull/16010 to Jetpack
* delete your site's full sync settings `delete_option( 'jetpack_sync_full_status' );`
* trigger cron to execute (do_action) or wp cron event run
* verify process loops to max execution time

Verify issue in 8.6.0 is resolved
* Apply this PR
* delete your site's full sync settings `delete_option( 'jetpack_sync_full_status' );`
* trigger cron to execute (do_action) or wp cron event run
* verify process loops once and exits.

Verify issue is not present in master
* checkout master
* Apply this PR
* delete your site's full sync settings `delete_option( 'jetpack_sync_full_status' );`
* trigger cron to execute (do_action) or wp cron event run
* verify process loops once and exits.

#### Proposed changelog entry for your changes:
* WP.com synchronization : hardening of cron flow to limit executions
